node/src/accountant/big_int_db_processor.rs

```diff
@@ -20,67 +19,120 @@ use std::marker::PhantomData;
-            .collect::<Vec<(&str, &dyn ToSql)>>();
+            .pure_rusqlite_params_with_wei_params((&config.params.wei_change_params).into());
         match stm.execute(params.as_slice()) {
             Ok(_) => Ok(()),
             //SQLITE_CONSTRAINT_DATATYPE (3091),
             //the moment of Sqlite trying to store the number as REAL in a strict INT column
             Err(Error::SqliteFailure(e, _)) if e.extended_code == 3091 => {
-                self.update_threatened_by_overflow(conn, config)
+                self.overflow_handler.update_with_overflow(conn, config)
             }
             Err(e) => Err(BigIntDbError(format!(
-                "Wei change: error after invalid {} command for {} of {} Wei to {} with error '{}'",
+                "Error from invalid {} command for {} table and change of {} Wei to '{} = {}' with error '{}'",
                 config.determine_command(),
                 T::table_name(),
                 config.balance_change(),
-                config.key_info().1,
+                config.params.table_unique_key_name,
+                config.key_param_value(),
                 e
             ))),
         }
     }
+}
+
+impl<T: TableNameDAO + 'static> Default for BigIntDbProcessor<T> {
+    fn default() -> BigIntDbProcessor<T> {
+        Self {
+            overflow_handler: Box::new(UpdateOverflowHandlerReal::default()),
+            phantom: Default::default(),
+        }
+    }
+}
 
-    fn update_threatened_by_overflow<'a>(
+impl<T: TableNameDAO> BigIntDbProcessor<T> {
+    fn prepare_statement<'a>(
+        form_of_conn: Either<&'a dyn ConnectionWrapper, &'a Transaction>,
+        sql: &'a str,
+    ) -> Statement<'a> {
+        match form_of_conn {
+            Either::Left(conn) => conn.prepare(sql),
+            Either::Right(tx) => tx.prepare(sql),
+        }
+        .expect("internal rusqlite error")
+    }
+}
+
+pub trait UpdateOverflowHandler<T>: Debug + Send {
+    fn update_with_overflow<'a>(
+        &self,
+        conn: Either<&dyn ConnectionWrapper, &Transaction>,
+        config: BigIntSqlConfig<'a, T>,
+    ) -> Result<(), BigIntDbError>;
+}
+
+#[derive(Debug)]
+struct UpdateOverflowHandlerReal<T: TableNameDAO> {
+    phantom: PhantomData<T>,
+}
+
+impl<T: TableNameDAO> Default for UpdateOverflowHandlerReal<T> {
+    fn default() -> Self {
+        Self {
+            phantom: Default::default(),
+        }
+    }
+}
+
+impl<T: TableNameDAO> UpdateOverflowHandler<T> for UpdateOverflowHandlerReal<T> {
+    fn update_with_overflow<'a>(
         &self,
         conn: Either<&dyn ConnectionWrapper, &Transaction>,
         config: BigIntSqlConfig<'a, T>,
     ) -> Result<(), BigIntDbError> {
         let select_sql = config.select_sql();
-        let mut select_stm = Self::prepare_statement(conn, &select_sql);
+        let mut select_stm = BigIntDbProcessor::<T>::prepare_statement(conn, &select_sql);
         match select_stm.query_row([], |row| {
```

Comment:
Could you use a temporary result here so that we don't have a closure definition in the middle of a match? It's kind of hard to read this way. Maybe either declare the closure separately, or call .query_row() and get a result, then use match on the result?